cosmetics-web/app/controllers/poison_centres/notifications_controller.rb

```diff
@@ -18,11 +19,15 @@ def show
 private
 
   def search_notifications(page_size)
-    query = ElasticsearchQuery.new(query_params[:q])
+    query = ElasticsearchQuery.new(@form.q, @form.category)
     Notification.full_search(query).paginate(page: params[:page], per_page: page_size)
   end
 
-  def query_params
-    params.permit(:q)
+  def search_params
+    if params[:notification_search_form].present?
+      params.require(:notification_search_form).permit(:q, :category)
+    else
+      {}
+    end
```

Comment:
Suggesting for consistency with how is done through other param methods:
```suggestion
    params.fetch(:notification_search_form, {}).permit(:q, :category)
```
Maybe it doesn't work with the `@form = NotificationSearchForm.new(search_params)` initialisation and you discarded that approach though.